internal/services/automation/automation_connection_type_resource.go

```diff
@@ -0,0 +1,211 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/automation/mgmt/2020-01-13-preview/automation"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/automation/2021-06-22/automationaccount"
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/automation/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type FieldDefinition struct {
+	Name        string `tfschema:"name"`
+	IsOptional  bool   `tfschema:"is_optional"`
+	IsEncrypted bool   `tfschema:"is_encrypted"`
+	Type        string `tfschema:"type"`
+}
+
+type AutomationConnectionTypeModel struct {
+	ResourceGrup          string            `json:"resource_grup" tfschema:"resource_group_name"`
+	AutomationAccountName string            `json:"automation_account_name" tfschema:"automation_account_name"`
+	Name                  string            `json:"name" tfschema:"name"`
+	IsGlobal              bool              `json:"is_global" tfschema:"is_global"`
+	FieldDefinitions      []FieldDefinition `json:"field_definitions" tfschema:"field_definitions"`
+}
+
+type AutomationConnectionTypeResource struct{}
+
+var _ sdk.Resource = (*AutomationConnectionTypeResource)(nil)
+
+func (m AutomationConnectionTypeResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"automation_account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AutomationAccount(),
+		},
+
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ConnectionTypeName,
+		},
+
+		"is_global": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+		},
+
+		"field_definitions": {
```

Comment:
definitions feels redundant?
```suggestion
		"field": {
```